### PR TITLE
Update obstacle distance

### PR DESCRIPTION
**Pull request by mrivi**
@TSC21 it should address your concerns.
@julianoes is it okay?

### CODE REVIEW

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -4400,9 +4400,11 @@
     <message id="330" name="OBSTACLE_DISTANCE">
       <description>Obstacle distances in front of the sensor, starting from the left in increment degrees to the right</description>
       <field type="uint64_t" name="time_usec" units="us">Timestamp (microseconds since system boot or since UNIX epoch)</field>
-      <field type="uint8_t" name="estimator_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
+      <field type="uint8_t" name="sensor_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
       <field type="uint8_t[12]" name="distances">Distance of obstacles in front of the sensor starting on the left side. A value of 0 means no obstacle. In a array element, each unit corresponds to 1m.</field>
```
Comment (TSC21) [inline message_definitions/v1.0/common.xml]: should this be `uint16_t[12]`?

**Comment by julianoes** [comment]
I think that's ok since it's quite new and no QGC release has been made with this.

**Comment by TSC21** [comment]
There's also one thing I think we are missing here: one thing is having measurement = 0, meaning that there is no obstacle. But there there's a case I guess where the beam on a certain direction is not used/not activated (assuming we can use this msg for a group of sensors instead of a TOF type). And in that case we will want to state which sensor is reading and which is not. For that purpose, I suggest:
1. Changing the type to `uint16_t[12]`, for increased precision;
2. No obstacle, then `distances[i] = 0`, no beam/not used, `distances[i] = UINT16_MAX`. Should be referenced on the filed description on that case.

**Comment by julianoes** [comment]
@TSC21 I agree that we should have a special value for not used but we should also have some value for "unknown".

**Comment by TSC21** [comment]
> Well, on second thought, that could be the same.

Yeah, `UNKNOWN` or `NA` or `NOTUSED`. Turn out to be the same.

**Comment by mrivi** [comment]
@TSC21 @julianoes since we are refactoring a bit, what do you think about increasing the number of elements of the array distances? Now we have 12 elements so if we have a setup to have distance data 360 degrees we can only have a resolution of 30 degrees.

**Comment by TSC21** [comment]
> @TSC21 @julianoes since we are refactoring a bit, what do you think about increasing the number of elements of the array distances? Now we have 12 elements so if we have a setup to have distance data 360 degrees we can only have a resolution of 30 degrees.

Yeah I guess you can increase till you fit the payload of the message.

**Comment by mrivi** [comment]
I have increased the array size to 72 to have a potential resolution of 5 degrees

**Comment by TSC21** [comment]
> I have increased the array size to 72 to have a potential resolution of 5 degrees

Is the increment higher or lower than this?

**Comment by mrivi** [comment]
Lower, it would be 5 if you use all the 72 bins. If the increment is 10 the  you would need only 36 bins, etc..

**Comment by TSC21** [comment]
Yeah I guess that depends on the developer to set that accordingly.

**Comment by julianoes** [comment]
So what is the distance if you are basically bumping into an object? 0 and therefore no obstacle?

**Comment by mrivi** [comment]
For stereo cameras once you're closer to an obstacle than `min_distance` you won't have any indicative data of you're actual distance to the obstacle

**Comment by TSC21** [comment]
Yah basically you can use the min_distance as a validation value for the readings. Actually I'm not seeing any sensor that currently can give you a minimum range of 0.0.

**Comment by julianoes** [comment]
What if a new sensor comes around that claims it can do 0? Sorry if I'm going into an edge case here, I just find the choice for "no obstacle" of 0 weird.

**Comment by dogmaphobic** [comment]
>I just find the choice for "no obstacle" of 0 weird.

What number will it show right when you hit it? :)

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -4400,9 +4400,11 @@
     <message id="330" name="OBSTACLE_DISTANCE">
       <description>Obstacle distances in front of the sensor, starting from the left in increment degrees to the right</description>
       <field type="uint64_t" name="time_usec" units="us">Timestamp (microseconds since system boot or since UNIX epoch)</field>
-      <field type="uint8_t" name="estimator_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
-      <field type="uint8_t[12]" name="distances">Distance of obstacles in front of the sensor starting on the left side. A value of 0 means no obstacle. In a array element, each unit corresponds to 1m.</field>
+      <field type="uint8_t" name="sensor_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
+      <field type="uint16_t[72]" name="distances">Distance of obstacles in front of the sensor starting on the left side. A value of 0 means no obstacle. A value of UINT16_MAX for unknown/not used. In a array element, each unit corresponds to 1m.</field>
```
Comment (LorenzMeier) [inline message_definitions/v1.0/common.xml]: Let's use uint16 here and use 1 cm as unit. That would make it 650m maximum distance, which is probably reasonable.

**Comment by LorenzMeier** [comment]
I would propose the following:

  * 1 unit = 1 cm
  * 0 = obstacle is right in front of sensor
  * max_range + 1: no obstacle (positively confirmed)
  * UINT16_MAX: no measurement

**Comment by mrivi** [comment]
@LorenzMeier @julianoes I have addressed your comments.

**Pull request merged**